### PR TITLE
new random password script

### PR DESCRIPTION
**Pull request by Sybrand**
previous algorithm was generating characters that confuses hasura

e.g. `"postgresql://harmony-admin:HiekyClDY7M+ew/+qckLLMk+BMPi3o2bBjIR1RaRhfE=@postgres.harmony_network:5432/harmony"`
results in `{"code":"postgres-error","error":"connection error","internal":"invalid integer value \"HiekyClDY7M+ew\" for connection option \"port\"\n","path":"$"}`

### CODE REVIEW

**Comment by AbbyBeeler** [comment]
@Sybrand what was the Hasura error it was giving? I don't recall seeing this issue in our repository

**Comment by Ideonvn** [comment]
> > @Sybrand what was the Hasura error it was giving? I don't recall seeing this issue in our repository
> 
> The GRAPHQL_DATABASE_URL doesn't get parsed correctly by Hasura. Specifically if there are + and / characters - e.g. the connection string `postgresql://harmony-admin:HiekyClDY7M+ew/+qckLLMk+BMPi3o2bBjIR1RaRhfE=@postgres.harmony_network:5432/harmony` results in `{"code":"postgres-error","error":"connection error","internal":"invalid integer value \"HiekyClDY7M+ew\" for connection option \"port\"\n","path":"$"}` I've tried url encoding it, but it still breaks. Regardless, one would want the connection string that gets spat out to work out of the box.

@Sybrand one workaround (instead of changing the password generating algo) is to just encode the password when passing to Hasura. I ran into this issue before setting up Harmony and got around by doing this.

However not against changing the algo at all, so it is either or really - it would however we nice if we can include the option for someone to just pass in their own master password.

**Comment by Sybrand** [comment]
> > > @Sybrand what was the Hasura error it was giving? I don't recall seeing this issue in our repository
> > 
> > 
> > The GRAPHQL_DATABASE_URL doesn't get parsed correctly by Hasura. Specifically if there are + and / characters - e.g. the connection string `postgresql://harmony-admin:HiekyClDY7M+ew/+qckLLMk+BMPi3o2bBjIR1RaRhfE=@postgres.harmony_network:5432/harmony` results in `{"code":"postgres-error","error":"connection error","internal":"invalid integer value \"HiekyClDY7M+ew\" for connection option \"port\"\n","path":"$"}` I've tried url encoding it, but it still breaks. Regardless, one would want the connection string that gets spat out to work out of the box.
> 
> @Sybrand one workaround (instead of changing the password generating algo) is to just encode the password when passing to Hasura. I ran into this issue before setting up Harmony and got around by doing this.
> 
> However not against changing the algo at all, so it is either or really - it would however we nice if we can include the option for someone to just pass in their own master password.

Encoding the password is all good and well (though, I tried a few different encodings and it kept on pegging) - but it really should work out of the box. I don't want to pepper the README with additional instructions on how to encode the connection string we just generated for you.

Agreed re. specifying own password - but that's outside the scope of this PR.

**Comment by Ideonvn** [comment]
> > > > @Sybrand what was the Hasura error it was giving? I don't recall seeing this issue in our repository
> > > 
> > > 
> > > The GRAPHQL_DATABASE_URL doesn't get parsed correctly by Hasura. Specifically if there are + and / characters - e.g. the connection string `postgresql://harmony-admin:HiekyClDY7M+ew/+qckLLMk+BMPi3o2bBjIR1RaRhfE=@postgres.harmony_network:5432/harmony` results in `{"code":"postgres-error","error":"connection error","internal":"invalid integer value \"HiekyClDY7M+ew\" for connection option \"port\"\n","path":"$"}` I've tried url encoding it, but it still breaks. Regardless, one would want the connection string that gets spat out to work out of the box.
> > 
> > 
> > @Sybrand one workaround (instead of changing the password generating algo) is to just encode the password when passing to Hasura. I ran into this issue before setting up Harmony and got around by doing this.
> > However not against changing the algo at all, so it is either or really - it would however we nice if we can include the option for someone to just pass in their own master password.
> 
> Encoding the password is all good and well (though, I tried a few different encodings and it kept on pegging) - but it really should work out of the box. I don't want to pepper the README with additional instructions on how to encode the connection string we just generated for you.
> 
> Agreed re. specifying own password - but that's outside the scope of this PR.

URLencode worked for me on the first to, but generally yes it is a nightmare. I just tried to keep initial changes minimal.

Perhaps a tech dept task where we prompt the user for a password, if none is provided we generate one.

File: scripts/create_deployment_database.sh
Comment (jbinary) [reply scripts/create_deployment_database.sh:31]: I personally use `pwgen` but it kind of should be installed in order to be used.

Comment (Sybrand) [reply scripts/create_deployment_database.sh:31]: I don't want to add any new dependencies - this approach uses everything out of the box.

**Review by tahitihat — APPROVED** [review]
I like the idea of reducing instructions — for now — and autogenerating the pw. This looks fine to me although I don't know much about Linux password generation options so I will defer to others (consider waiting for a second approval). I added the prompt-user tech debt task to our internal Harmony backlog 👍

**Pull request merged**
